src/palace/manager/api/axis.py

```diff
@@ -498,6 +508,12 @@ def fulfill(
             return fulfillment
         # If we made it to this point, the patron does not have this
         # book checked out.
+        debug_info.insert(
+            0,
+            "Unable to fulfill because there is no active loan. See info statements below for details:",
+        )
+
+        self.log.debug("\n  ".join(debug_info))
```

Comment:
Sorry @dbernstein, I should have been more clear in my comment before. I think logging at the warning or even error level makes sense here, like you had before. Debug level logging is tricky to enable in production because of the huge volume of logs it generates. 

I wasn't very clear in my previous comment, but I was suggesting that you might want to change the variable name to something else, like `logging_info` or `log_messages`.

Reply:
On second thought, I think error is the way to go since in almost every situation, if not every situation it arises, it would indicate a real problem.